onetl/_util/version.py

```diff
@@ -2,193 +2,161 @@
 # SPDX-License-Identifier: Apache-2.0
 from __future__ import annotations
 
+import re
 from functools import total_ordering
-from string import ascii_lowercase
-from typing import Iterator, NamedTuple, Sequence
 
 
 @total_ordering
-class Version(NamedTuple):
+class Version:
     """
     Version representation.
 
     Examples
     --------
 
-    >>> Version(5, 6, 7)
-    Version(major=5, minor=6, patch=7)
-    >>> Version(5, 6)
-    Version(major=5, minor=6, patch=None)
-    >>> Version(5)
-    Version(major=5, minor=None, patch=None)
+    >>> Version("12.3.1")
+    Version('12.3.1')
+    >>> Version("12.3")
+    Version('12.3')
 
     """
 
-    major: int
-    minor: int | None = None
-    patch: int | None = None
-
-    def __iter__(self) -> Iterator[int]:
+    def __init__(self, version):
+        if isinstance(version, Version):
+            self._raw_str = version._raw_str
+            self._raw_parts = version._raw_parts.copy()
+            self._numeric_parts = version._numeric_parts.copy()
+        elif isinstance(version, str):
+            self._raw_str = version
+            self._raw_parts = re.split("[.-]", version)
+            self._numeric_parts = []
+
+            # process parts to keep leading zeros in minor or patch levels (like '00' in '12.23.00.34')
+            for part in self._raw_parts:
+                clean_part = "".join(filter(str.isdigit, part)) or "0"
+                self._numeric_parts.append(int(clean_part))
+
+    @property
+    def major(self) -> int:
         """
-        Iterate over version components which are not ``None``.
+        Return the major version component.
 
         Examples
         --------
 
-        >>> for part in Version(5, 6, 7):
-        ...     print(part)
-        5
-        6
-        7
-
-        >>> for part in Version(5, 6):
-        ...     print(part)
-        5
-        6
-
-        >>> for part in Version(5):
-        ...     print(part)
-        5
-
+        >>> assert Version("5.6.7").major == 5
```

Comment:
```suggestion
        >>> Version("5.6.7").major
        5
```

Давай попробуем убрать assert из docstring